robot_smach_states/src/robot_smach_states/world_model/world_model.py

```diff
@@ -215,6 +215,7 @@ def execute(self, userdata=None):
                 rospy.logerr("    Classification failed, this should not happen!")
         else:
             rospy.logwarn(">> Tried to segment but no objects found")
+            self.segmented_entity_ids_designator.write([])
```

Comment:
Based on the commit I  assume this was done because the result of this is used by the designator iterator state. Though I think we should keep the result a None and let the iterator state handle None's correctly.

Reply:
No, I think it is failed to set an empty list when nothing found.